src/decorators/registry.ts

```diff
@@ -0,0 +1,27 @@
+import { handleDecorator, DecoratorHandler } from './handleDecorator';
+import { RegistryItem } from '../Registry';
+
+export interface RegistryConfig {
+	[name: string]: RegistryItem;
+}
+
+/**
+ * Decorator that can be used to register a widget with the calling widgets local registry
+ */
+export function registry(nameOrConfig: string, loader: RegistryItem): DecoratorHandler;
```

Comment:
@matt-gadd when we talked about this before, did we decide not to use `registry` but `define` as the decorators name?

Reply:
`define` gets picked up by webpack according to @matt-gadd , so that's a negatory

Reply:
`@define` gets picked up by webpack? In what way?

Reply:
@agubler reserved word pretty much for AMD.